src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java

```diff
@@ -39,32 +50,107 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
+
+        if (isReadyToList(postponementRequest)) {
+            sscsCaseData.setState(State.READY_TO_LIST);
+        } else if (isNotListable(postponementRequest)) {
+            sscsCaseData.setState(State.NOT_LISTABLE);
+        }
+
+        updatePartyUnavailability(sscsCaseData);
+
+        sscsCaseData.setInterlocReferralReason(null);
+        sscsCaseData.setInterlocReviewState(null);
+        sscsCaseData.setDwpState(DwpState.DIRECTION_ACTION_REQUIRED.getId());
+        addDirectionNotice(sscsCaseData);
+        sscsCaseData.setPostponementRequest(PostponementRequest.builder().unprocessedPostponementRequest(YesNo.NO)
+                .build());
+    }
+
+    private void updatePartyUnavailability(SscsCaseData sscsCaseData) {
+        Optional<Hearing> futureHearing = sscsCaseData.getHearings().stream()
+                .filter(hearing -> getLocalDate(hearing.getValue().getHearingDate()).isAfter(LocalDate.now()))
+                .findAny();
+
+        if (futureHearing.isPresent()) {
+            String hearingDateString = getLocalDate(futureHearing.get().getValue().getHearingDate())
+                    .format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+            DateRange dateRange = DateRange.builder()
+                    .start(hearingDateString)
+                    .end(hearingDateString)
+                    .build();
+            HearingOptions hearingOptions = sscsCaseData.getAppeal().getHearingOptions();
+            List<ExcludeDate> excludedDates = mutableEmptyListIfNull(hearingOptions.getExcludeDates());
+            excludedDates.add(ExcludeDate.builder().value(dateRange).build());
+            hearingOptions.setExcludeDates(excludedDates);
+        }
+    }
+
+    private void sendToJudge(String userAuthorisation, SscsCaseData sscsCaseData) {
+        if (sscsCaseData.getAppealNotePad() == null) {
+            sscsCaseData.setAppealNotePad(NotePad.builder().notesCollection(new ArrayList<>()).build());
+        }
+        sscsCaseData.getAppealNotePad().getNotesCollection()
+                .add(createPostponementRequestNote(userAuthorisation,
+                        sscsCaseData.getPostponementRequest().getPostponementRequestDetails()));
+        sscsCaseData.setInterlocReviewState(InterlocReviewState.REVIEW_BY_JUDGE.getId());
+        sscsCaseData.setInterlocReferralReason(InterlocReferralReason.REVIEW_POSTPONEMENT_REQUEST.getId());
+        sscsCaseData.setPostponementRequest(PostponementRequest.builder().unprocessedPostponementRequest(YesNo.YES)
+                .build());
+    }
+
+    private void addDirectionNotice(SscsCaseData caseData) {
+        DocumentLink url = caseData.getPreviewDocument();
+        SscsDocumentTranslationStatus documentTranslationStatus = caseData.isLanguagePreferenceWelsh() ? SscsDocumentTranslationStatus.TRANSLATION_REQUIRED : null;
+        footerService.createFooterAndAddDocToCase(url, caseData, AUDIO_VIDEO_EVIDENCE_DIRECTION_NOTICE,
```

Comment:
This is wrong event type. Has to be ACTION_POSTPONEMENT_REQUEST

Reply:
Well spotted. We already have POSTPONEMENT_REQUEST though so I am creating a new type for the directions notice. Nice one :)